### ISSUE TITLE
PHP Analyzer - alternative return typehints

### ISSUE THREAD

**Issue by pamil** [scrutinizer-ci/php-analyzer#708]
Documentation of parent function says:

```php
    /**
     * Returns whether the constraint can be put onto classes, properties or
     * both.
     *
     * This method should return one or more of the constants
     * Constraint::CLASS_CONSTRAINT and Constraint::PROPERTY_CONSTRAINT.
     *
     * @return string|array One or more constant values
     *
     * @api
     */
```

PHP Analyzer seems not to care about alternative return types.

See https://scrutinizer-ci.com/g/Sylius/Sylius/inspections/4706879a-fbd7-4d76-8177-f95687f9e291/issues/files/src/Sylius/Bundle/ResourceBundle/Validator/Constraints/Disabled.php?status=new&orderField=path&order=asc&issueId=9ba22efe-a78d-4495-8bfc-8e8452dafca5
